web/source/osk/visualKeyboard.ts

```diff
@@ -1491,17 +1518,29 @@ namespace com.keyman.osk {
       kbd.style.border = '1px solid #ccc';
 
       // Once the element is inserted into the DOM, refresh the layout so that proper text scaling may apply.
-      const refreshInterval = window.setInterval(function () {
-        let computedStyle = getComputedStyle(kbd);
-        if (computedStyle.fontSize) {
-          if (kbd.style.fontSize) {
-            // Preserve the new setting (provided by CSS)
-            kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+      const detectAndHandleInsertion = () => {
+        if(document.contains(kbd)) {
+          // Yay, insertion!
+
+          try {
+            // Are there font-size attributes we may safely adjust?  If so, do that!
+            if(getComputedStyle(kbd) && kbd.style.fontSize) {
+              kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+            }
+
+            // Make sure that the stylesheet is attached, now that the keyboard-doc's been inserted.
+            kbdObj.appendStyleSheet();
```

Comment:
Either way, re: `MutationObserver` vs `repeatInterval`, this section is the right place to ensure that the keyboard's stylesheet is properly appended.